From 8f0d617efc16dda3689d3e13f8e9f000f6502266 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 14 Nov 2018 15:06:12 +0200
Subject: [PATCH 5140/5242] MLK-20216-6: video/fbdev/mxc_dcic: fix return
 value

commit  228eebe7be95208049439b37d95cd1f732897152 from
https://source.codeaurora.org/external/imx/linux-imx.git

This fixes Coverity issue where an unsigned variable is tested for
less-than-zero value.

Make val a signed integer instead.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
CC: Sandor Yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_dcic.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mxc_dcic.c b/drivers/video/fbdev/mxc/mxc_dcic.c
index 3c5a83be..3e6824a 100644
--- a/drivers/video/fbdev/mxc/mxc_dcic.c
+++ b/drivers/video/fbdev/mxc/mxc_dcic.c
@@ -360,7 +360,8 @@ static int dcic_release(struct inode *inode, struct file *file)
 
 static int dcic_init(struct device_node *np, struct dcic_data *dcic)
 {
-	u32 val, bus;
+	int val;
+	u32 bus;
 
 	val = of_get_dcic_val(np, dcic);
 	if (val < 0) {
-- 
1.7.9.5

